Problem: Running new npm tasks fails using qa/1.7.x

When trying to deploy the new appraisal tab and transfer browser features I found an issue when running the npm tasks. It was trying to run the npm tasks as user root, using /root/.npm directory.

I could deploy this tasks as archivematica user, although I had to change the owner of /opt/archivematica previously.

The following code was used in the tasks/pipeline-instcode.yml file:

```
#
# front-end
#

#npm needs to be running as user archivematica
#The owner of archivematica_src_dir has to be changed to archivematica
- name: "Change archivematica-source owner to archivematica"
  file:
    dest: "{{ archivematica_src_dir }}"
    state: "directory"
    owner: "archivematica"
    group: "archivematica"
    recurse: "yes"

- name: "Install front-end dependencies"
  become: "yes"
  become_user: "archivematica"
  npm:
    path: "{{ item }}"
    state: "present"
  with_items:
    - "{{ archivematica_src_dir }}/archivematica/src/dashboard/frontend/appraisal-tab"
    - "{{ archivematica_src_dir }}/archivematica/src/dashboard/frontend/transfer-browser"
  when:
    - "ansible_env.USER != 'vagrant'"
```

I don't know if it is better to change the archivematica_src_dir owner at another point of the deployment.

This change works for our deployment on Ubuntu 14.04 lts as well.

However setting `dest` set to  `{{ archivematica_src_dir }}/src/dashboard/src/media` throws a 'npm ERR! Error: EACCES: permission denied'

--- a/tasks/pipeline-instcode.yml
+++ b/tasks/pipeline-instcode.yml
@@ -82,7 +82,20 @@
 # front-end
 #
 
+- name: "Change archivematica-source owner to archivematica"
+  file:
+    dest: "{{ archivematica_src_dir }}"
+    state: "directory"
+    owner: "archivematica"
+    group: "archivematica"
+    recurse: "yes"
+  with_items:
+    - "{{ archivematica_src_dir }}/archivematica/src/dashboard/src/media"
+    - "{{ archivematica_src_dir }}/archivematica/src/dashboard/frontend"
+
 - name: "Install front-end dependencies"
+  become: "yes"
+  become_user: "archivematica"
   npm:
     path: "{{ item }}"
     state: "present"
@@ -90,6 +103,9 @@
     - "{{ archivematica_src_dir }}/archivematica/src/dashboard/frontend/appraisal-tab"
     - "{{ archivematica_src_dir }}/archivematica/src/dashboard/frontend/transfer-browser"
   when:
+    # We're not building the front-ends when using Vagrant because the source
+    # code lives in a shared folder (vboxsf), which is problematic and causes
+    # errors. The developer needs to build the front-ends manually for now.
     - "ansible_env.USER != 'vagrant'"
 
 

